behaviour/graql/language/match.feature

```diff
@@ -191,68 +183,897 @@ Feature: Graql Match Clause
+      person has graduation-date;
+      employment has start-date;
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "James", has ref 0, has graduation-date 2009-07-16;
+      $r (employee: $x) isa employment, has start-date 2009-07-16, has ref 1;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match
+        $x isa person, has graduation-date $date;
+        $r (employee: $x) isa employment, has start-date $date;
+      get;
+      """
+    Then answer size is: 0
+    Then get answers of graql query
+      """
+      match
+        $x isa person, has graduation-date $date;
+        $r (employee: $x) isa employment, has start-date == $date;
+      get;
+      """
+    Then answer size is: 1
+
+
+  Scenario: `has $attr > $x` matches owners of any instance `$y` of `$attr` where `$y > $x`
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Susie", has age 16, has ref 0;
+      $y isa person, has name "Donald", has age 25, has ref 1;
+      $z isa person, has name "Ralph", has age 18, has ref 2;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has age > 18; get;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | DON | key   | ref:1 |
+    Then uniquely identify answer concepts
+      | x   |
+      | DON |
+
+
+  Scenario: `has $attr < $x` matches owners of any instance `$y` of `$attr` where `$y < $x`
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Susie", has age 16, has ref 0;
+      $y isa person, has name "Donald", has age 25, has ref 1;
+      $z isa person, has name "Ralph", has age 18, has ref 2;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has age < 18; get;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | SUS | key   | ref:0 |
+    Then uniquely identify answer concepts
+      | x   |
+      | SUS |
+
+
+  Scenario: `has $attr !== $x` matches owners of any instance `$y` of `$attr` where `$y !== $x`
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Susie", has age 16, has ref 0;
+      $y isa person, has name "Donald", has age 25, has ref 1;
+      $z isa person, has name "Ralph", has age 18, has ref 2;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has age !== 18; get;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | DON | key   | ref:1 |
+      | SUS | key   | ref:0 |
+    Then uniquely identify answer concepts
+      | x   |
+      | DON |
+      | SUS |
+
+
+  Scenario: `has $attr > $x` matches owners of any instance `$y` of `$attr` where `$y > $x` even if they also own instance `$z` where `$z < $x`
```

Comment:
probably like `attribute value comparison returns a matching subset of all attributes owned by a thing`

Reply:
How about `when a thing owns multiple attributes of the same type, a value comparison matches if any value matches`? @flyingsilverfin